sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/FileScan.scala

```diff
@@ -41,7 +41,7 @@ abstract class FileScan(
     val selectedPartitions = fileIndex.listFiles(Seq.empty, Seq.empty)
     val maxSplitBytes = FilePartition.maxSplitBytes(sparkSession, selectedPartitions)
     val splitFiles = selectedPartitions.flatMap { partition =>
-      partition.files.flatMap { file =>
+      partition.files.filter(_.getLen > 0).flatMap { file =>
```

Comment:
Don't forget to remove this.

Reply:
@MaxGekk Thanks. This is code changes from https://github.com/apache/spark/pull/24227 and I will remove it once https://github.com/apache/spark/pull/24227 is merged.